### ISSUE TITLE
Possible memleak detected

### ISSUE THREAD

**Issue by JBustin** [nuxt/nuxt#30313]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.10.0
- Nuxt Version:     3.14.1592
- CLI Version:      3.16.0
- Nitro Version:    2.10.4
- Package Manager:  npm@10.2.5
- Builder:          -
- User Config:      default
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://github.com/JBustin/nuxt3-mem-leak-repro

### Describe the bug

When I log the runtime config with the console.log function in a component, the runtime config content is duplicated in nodejs memory for each console.log execution.

### Additional context

I have a big application that I'm migrating from nuxt2 to nuxt3 and I'm experimenting a very bad behavior with memory.
A lot of things are duplicated in SSR memory: runtime config, dom snapshots, chunks, ...
<img width="1444" alt="Image" src="https://github.com/user-attachments/assets/5de3fbfa-355b-4db1-bd5f-fcda65807a4d" />
I'm searching in my code, and, in the same time, I'm testing nuxt3 in different situations.
I hope that the memory issue that I reproduce with the nuxt3 starter could help me to understand what happens.

**Comment by danielroe** [comment]
This is true when debugging - any logging to console of an object or array will keep that object in memory for the duration of the debugging session (because it's available in the console for you to click into, inspect, etc.).

But it's an artifact of the debugging itself, not an indication of a memory leak.

**Comment by JBustin** [comment]
Hello @danielroe 
Thanks for your response.
I update the test to console.log an object, and I don't have the same behavior for this second object, no duplicate in memory.
```html
<script setup lang="ts">
const config = useRuntimeConfig()
// persistent duplicated public apiKey in memory, one by code execution
console.log(config)

// no duplicates
const test = { value: 'COUCOU' }
console.log(test)
</script>
```

**Comment by danielroe** [comment]
can you update your reproduction with no console.logging, and make sure to garbage collect before checking memory usage?

if it still reproduces for you, let me know and I'll have a look

**Comment by JBustin** [comment]
Hello @danielroe, without the console.log, there is no duplicate of runtime config in the inspector, but, there is a slight but continued increase in memory, that I can see after each load test.
After each load tests, during the idle time, memory slightly increases (regarding the previous idle time): 187M, then 193M, 196M, 198M, 200M, ... after more than 10 tests, it doesn't stop growing.
I'm just testing the bootstrap with a runtime config as you can see in my github example.

**Comment by danielroe** [comment]
@JBustin are you garbage collecting before measuring? by design node increases memory usage until it hits the limit

**Comment by JBustin** [comment]
I assume that the big decrease of memory after each test is due to the garbage collector, I wait this step before taking the measure. The shape of memory is like I share 3 weeks ago: https://github.com/nuxt/nuxt/issues/30313#issuecomment-2556461298

**Comment by JBustin** [comment]
https://github.com/JBustin/nuxt3-mem-leak-repro/tree/no-code-test

The diff may seem insignificant, but personally, I will be reassured when it stops increasing ^^
If you're sure it's garbage collector behavior, perhaps of some resources managed by nodejs, waiting for a certain limit to be reached, no problem.
For info, I'm experimenting the same behavior with a more elaborate application (runtime config + scss + custom composables + svg + ...), but with some higher diff between each tests, like between 3 and 10M of additional memory.

**Comment by egobude** [comment]
Hi everyone,

We're currently experiencing the same issue described here. This is causing significant challenges for us. If there are any approaches or workarounds available, we would greatly appreciate your insights.

Thanks for your support!

**Comment by mstd3v** [comment]
Hey there! 
My team and I are currently facing the same problem. 
What's interesting is that when we take snapshots during our local debugging via DevTools, we can see that a lot (and I mean a LOT) of duplicates (especially strings) stay in memory for at least the length of the test also used in @JBustin's example. 

Here you can see how the memory size increases exponentially during the execution of the above test:
![Image](https://github.com/user-attachments/assets/53e0125d-c111-4c3c-80c1-5011680a982d)

We tried multiple nuxt versions (including the latest) with the same behaviour.
Would you still consider these numbers as expected behaviour or is there anything we can do about that?

Thank you so much in advance!

**Comment by danielroe** [comment]
@mstd3v That would be unusual if you were garbage collecting before each snapshot, and you were running a _built_ Nuxt app...

**Comment by BobbieGoede** [comment]
The reproduction project repositories are no longer accessible (private or removed), if someone could provide a minimal reproduction + steps to reproduce the leak, I'd be happy to look into this.
